contract/models/contract.py

```diff
@@ -327,59 +327,34 @@ def action_contract_send(self):
             'target': 'new',
             'context': ctx,
         }
-
     @api.model
     def _finalize_invoice_values(self, invoice_values):
-        """
-        This method adds the missing values in the invoice lines dictionaries.
-
-        If no account on the product, the invoice lines account is
-        taken from the invoice's journal in _onchange_product_id
-        This code is not in finalize_creation_from_contract because it's
-        not possible to create an invoice line with no account
-
-        :param invoice_values: dictionary (invoice values)
-        :return: updated dictionary (invoice values)
-        """
-        # If no account on the product, the invoice lines account is
-        # taken from the invoice's journal in _onchange_product_id
-        # This code is not in finalize_creation_from_contract because it's
-        # not possible to create an invoice line with no account
-        new_invoice = self.env['account.invoice'].with_context(
-            force_company=invoice_values['company_id'],
-        ).new(invoice_values)
-        for invoice_line in new_invoice.invoice_line_ids:
-            name = invoice_line.name
-            account_analytic_id = invoice_line.account_analytic_id
-            price_unit = invoice_line.price_unit
-            invoice_line.invoice_id = new_invoice
-            invoice_line._onchange_product_id()
-            invoice_line.update(
-                {
-                    'name': name,
-                    'account_analytic_id': account_analytic_id,
-                    'price_unit': price_unit,
-                }
-            )
-        return new_invoice._convert_to_write(new_invoice._cache)
+        """Provided for keeping compatibility in this version."""
+        # TODO: Must be removed in >=13.0
```

Comment:
nit: log a deprecation warning?

Reply:
I prefer not as we are going to keep it in this version, so deprecation doesn't apply at all, and when migrating then the TODO is clear. Populating the logger needs also several lines that doesn't worth IMO.